### PR TITLE
Amazon S3 bucket external access

### PR DESCRIPTION
**Pull request by jonathanglassman**
What
----

Added content on how to access the Amazon S3 bucket from outside of the GOV.UK PaaS.

How to review
-------------

Describe the steps required to test the changes.

1. Preview the content locally ([see README](https://github.com/alphagov/paas-tech-docs#preview)) and check that it renders as expected.
1. Manually check that any removed or renamed anchor tags are not in use ([linkchecker](https://github.com/linkcheck/linkchecker) doesn't do this).
1. Check it fulfils the https://www.pivotaltracker.com/story/show/166837820 story.

Who can review
--------------

Anyone except Jon Glassman or Tom Whitwell.

### CODE REVIEW

File: source/documentation/deploying_services/s3.md [hunk 1]
```diff
@@ -106,6 +110,51 @@ An example `VCAP_SERVICES` environment variable:
...
+    {
+     "aws_access_key_id": "AWS_ACCESS_KEY",
+     "aws_region": "eu-west-2",
+     "aws_secret_access_key": "AWS_SECRET_ACCESS_KEY",
+     "bucket_name": "BUCKET_NAME_GUID",
+     "deploy_env": ""
+    }
+    ```
+
+    Use these credentials to connect to the S3 bucket from outside of the GOV.UK PaaS.
```
Comment (tnwhitwell) [inline source/documentation/deploying_services/s3.md]: The only thing here is that I wonder if we should point out that the region specific **must** be used here, as we are specifically creating the buckets in the same region as the PaaS.

I'll need to do some digging to find out what that actually is though

File: source/documentation/deploying_services/s3.md [hunk 2]
```diff
@@ -106,6 +110,51 @@ An example `VCAP_SERVICES` environment variable:
...
+
+To connect to the S3 bucket from outside of the GOV.UK PaaS, you must create a service key and credentials that allow external access to the bucket.
+
+1. Run the following to create a service key and credentials that allows external access:
+
+    ```
+    cf create-service-key SERVICE_NAME SERVICE-KEY -c '{"allow_external_access": true}'
+    ```
+    where:
+    - `SERVICE_NAME` is a unique descriptive name for the S3 bucket
```
[the remark this answers is not in the document]
Comment (jonathanglassman) [reply source/documentation/deploying_services/s3.md]: agree

File: source/documentation/deploying_services/s3.md (hunk 1 above)
Comment (tnwhitwell) [reply source/documentation/deploying_services/s3.md]: not a valid concern: `aws s3 ls s3://BUCKET_NAME/` works A-OK

File: source/documentation/deploying_services/s3.md [hunk 3]
```diff
@@ -106,6 +110,51 @@ An example `VCAP_SERVICES` environment variable:
...
+    ```
+
+    You will see output similar to the following:
+
+    ```
+    {
+     "aws_access_key_id": "AWS_ACCESS_KEY",
+     "aws_region": "eu-west-2",
+     "aws_secret_access_key": "AWS_SECRET_ACCESS_KEY",
+     "bucket_name": "BUCKET_NAME_GUID",
```
Comment (tnwhitwell) [inline source/documentation/deploying_services/s3.md]:
```suggestion
     "bucket_name": "BUCKET_NAME",
```

**Comment by richardTowers** [comment]
Do we need a second pair of tech writer eyes? Or shall I merge?

**Comment by jonathanglassman** [comment]
> Do we need a second pair of tech writer eyes? Or shall I merge?

@richardTowers yes it needs 2i. I'll send on for 2i now.

**Pull request merged**